src/v/cluster/partition_probe.h

```diff
@@ -105,6 +123,8 @@ class replicated_partition_probe : public partition_probe::impl {
     uint64_t _bytes_fetched{0};
     uint64_t _bytes_fetched_from_follower{0};
     uint64_t _schema_id_validation_records_failed{0};
+    uint64_t _iceberg_translation_offset_lag{0};
+    uint64_t _iceberg_commit_offset_lag{0};
```

Comment:
Should we default to -1 for uninitialized?

Reply:
Also should we initialize this before we start translation instead of just after the first translation?

Reply:
> Also should we initialize this before we start translation instead of just after the first translation?

Initialize to what though? We get the data as the translator starts to reconcile with the coordinator.

Reply:
We only update after we call `do_translation_for_range`. I'm looking forward a bit when we have more dynamic batching and if we are delaying translation until there is more data I think it's useful that we set the lag the first time we start processing. It's probably not a big deal right now and we can wait until we see how this evolves with the more dynamic batching.

Reply:
ya we can probably revisit this a bit, just want to get something in for the beta.